content/Blog/2018-02-01-design-the-best-architecture-ever.md

```diff
@@ -0,0 +1,79 @@
+{
+    "title": "How to determine the system architecture?",
+    "date": "2018-02-01"
+}
+
+# SYSTEM ARCHITECTURE
+
+
+## What's an architecture ?
+
+A system can be considered as a group of subsystems with their own functionalities and goals.
+The architecture is both the general structure distribution of sub-modules on board, and the definition of their relations.
+
+We previously presented our 5 modules ADCS, TCS, OBC, EDT and EPS. Our architecture has to create subsystems in terms of hardware, communication and energy, while respecting spatial requirements and being the most failure-tolerant.
+
+----------
+
+## How to find the best architecture ? (methodology)
+For a long and ambitious project like ECE3SAT, we wished to follow embedded system methodology.
+We applied the process found in the book *Embedded Systems Architectures* described in the following article.
+
+#### 1. Having a solid technical base
+Our project covers various technical realms. The first aspect is to gather knowledge in order to understand most requirements.
+
+#### 2. Understanding the architecture business cycle
+First of all, we had to gather all requirements from each module. The previous phase focused on it and used formalism to describe it in their reports. However, some requirements had to be updated, that's why dialog with other modules (our final clients) was crucial.
+
+We kept in mind that any information has to be saved (even if it’s written on a napkin). Indeed, we have to be able to make our choices understandable by future working teams.
+
+#### 3. Deﬁning the architectural patterns and reference models
+A state of the art and a study of existing cubesat allows our team to quickly identify important modules : EPS, ADCS, OBC, and TCS.
+
+#### 4. Creating the architectural structures
+>This is done by decomposing the entire embedded system into hardware and/or software elements, and then further decomposing the elements that need breaking down. These decompositions are represented as some combination of various types of structures [...]. The patterns [...] that most satisfy the system requirements (the most complete, the most accurate, the most buildable, highest conceptual integrity, etc.) should be used as the foundations for the architectural structures.

```

Comment:
It would be nice to have a special style for quotes (currently this just looks like a shifted ordinary paragraph and it's weird...)
@nodraak ?